packages/core/src/modules/connections/repository/ConnectionMetadataTypes.ts

```diff
@@ -0,0 +1,9 @@
+export enum ConnectionMetadataKeys {
+  UseDidKeysForProtocol = '_internal/useDidKeysForProtocol',
+}
+
+export type ConnectionMetadata = {
+  [ConnectionMetadataKeys.UseDidKeysForProtocol]: {
+    [protocolUri: string]: boolean
```

Comment:
Specific reason for making this a key/value object instead of an array with string uri's?

Reply:
The goal of this metadata key is to override agent settings, based on what we learn from the connection. So if our agent is set to always use did:key by default and we learn that the other side is not using it for a certain protocol, we'll set the key for that protocol to false.

And we could also have the opposite case: we don't use did:key but it happens that the other side is using them, so we set the key for that protocol to true. Also we might change our agent settings afterwards, and it will still take into account the metadata key for the given connection.

Reply:
Makes sense. I thought we would only use it to say a protocol does support did:key, but there is also the case where we explicitly store a protocol doesn't store it.

Is this also updated autmoatically when this changes? E.g. we have stored false so we keep sending base58 keys, but then the other agent sends a did:key, do we update it automatically? Or is it set once and kept like this forever?

Reply:
> Is this also updated autmoatically when this changes? E.g. we have stored false so we keep sending base58 keys, but then the other agent sends a did:key, do we update it automatically? Or is it set once and kept like this forever?

I think it depends on each protocol/service but generally the best would be to update it automatically. As of current AFJ support for mediator role, it does not add so much value because we only support Mediation Grant and Keylist Update request/response. However, considering a future support of Keylist queries and responses it's better to handle it right from the start.

So I'm adding automatic handling for both recipient and mediator in the messages we currently support. Hopefully in 0.3.x we'll support all of them :smile: